### ISSUE TITLE
tests/provider: Avoid hardcoding service/endpoint ID

### ISSUE THREAD

**Issue by YakDriver** [hashicorp/terraform-provider-aws#15974]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The provider has the `testAccPartitionHasServicePreCheck()` function which allows checking a service ID for support within a partition. However, we (mostly me) have been hardcoding the service ID rather than using the `const` from the service package.

#### TL;DR

Change `testAccPartitionHasServicePreCheck("appmesh", t)` to `testAccPartitionHasServicePreCheck(appmesh.EndpointsID, t)`.

### Existing Reports

- [ ] appsync (#15975)
- [ ] budgets (#15976)
- [ ] cloud9 (#15977)
- [x] cloudfront (#15978)
- [x] codeartifact (#15979)
- [x] codestar (#15980)
- [x] fsx (#15981)
- [x] models.lex (#15983)
- [x] opsworks (#15984)
- [x] waf (#15985)
- [x] waf-regional (#15986)
- [x] Update contributing guide to include use of `EndpointsID`. (#15988)

### Definition of Done

* Fix existing reports
* Search for `testAccPartitionHasServicePreCheck("` returns 0 results

### References

* #0000
